main.tf

```diff
@@ -282,7 +292,9 @@ resource "azurerm_key_vault" "service" {
 resource "azurerm_storage_account" "service" {
   for_each = toset(var.regions)
 
-  name                     = "${local.service_name}${each.key}${local.environment_differentiator_short}${var.environment}"
+  name                     = format("${local.service_name}%s${local.environment_differentiator_short}${var.environment}", 
```

Comment:
Instead of the format we could use something like:

locals {
  region_map = var.use_region_shortcodes ? module.region_to_short_region[0].mapping : {}
}

${lookup(local.region_map, each.key)}

This should simplify the code quite a bit for all of the different names

Reply:
Can do!